crates/viewer/re_blueprint_tree/src/blueprint_tree.rs

```diff
@@ -868,46 +853,97 @@ impl BlueprintTree {
     let default_blueprint_id = ctx
         .store_context
         .hub
         .default_blueprint_id_for_app(&ctx.store_context.app_id);
 
     let default_blueprint = default_blueprint_id.and_then(|id| ctx.store_context.bundle.get(id));
 
-    let mut disabled_reason = None;
-
-    if let Some(default_blueprint) = default_blueprint {
-        let active_is_clone_of_default = Some(default_blueprint.store_id()).as_ref()
-            == ctx.store_context.blueprint.cloned_from();
-        let last_modified_at_the_same_time =
-            default_blueprint.latest_row_id() == ctx.store_context.blueprint.latest_row_id();
-        if active_is_clone_of_default && last_modified_at_the_same_time {
-            disabled_reason = Some("No modifications have been made");
+    let disabled_reason = match default_blueprint {
+        None => Some("No default blueprint is set for this app"),
+        Some(default_blueprint) => {
+            let active_is_clone_of_default = Some(default_blueprint.store_id()).as_ref()
+                == ctx.store_context.blueprint.cloned_from();
+            let last_modified_at_the_same_time =
+                default_blueprint.latest_row_id() == ctx.store_context.blueprint.latest_row_id();
+            if active_is_clone_of_default && last_modified_at_the_same_time {
+                Some("No modifications have been made")
+            } else {
+                None // it is valid to reset to default
+            }
         }
-    }
+    };
 
     let enabled = disabled_reason.is_none();
-    let response = ui.add_enabled(enabled, ui.small_icon_button_widget(&re_ui::icons::RESET));
-
-    let response = if let Some(disabled_reason) = disabled_reason {
-        response.on_disabled_hover_text(disabled_reason)
-    } else {
-        let hover_text = if default_blueprint_id.is_some() {
-            "Reset to the default blueprint for this app"
-        } else {
-            "Re-populate viewport with automatically chosen views"
-        };
-        response.on_hover_text(hover_text)
+    let mut response = ui
+        .add_enabled(
+            enabled,
+            egui::Button::image_and_text(&re_ui::icons::RESET, "Reset to default blueprint"),
+        )
+        .on_hover_text("Reset to the default blueprint for this app");
+
+    if let Some(disabled_reason) = disabled_reason {
+        response = response.on_disabled_hover_text(disabled_reason);
     };
 
     if response.clicked() {
+        ui.close_menu();
         ctx.command_sender
             .send_system(re_viewer_context::SystemCommand::ClearActiveBlueprint);
     }
 }
 
+fn set_blueprint_to_auto_menu_button(
+    ctx: &ViewerContext<'_>,
+    viewport: &ViewportBlueprint,
+    ui: &mut egui::Ui,
+) {
+    let enabled = !viewport.auto_layout() || !viewport.auto_views();
+
+    if ui
+        .add_enabled(
+            enabled,
+            egui::Button::image_and_text(&re_ui::icons::RESET, "Reset to heuristic blueprint"),
```

Comment:
I wonder if we can come up with better copy here. "Create automatic blueprint"? Maybe @gavrelina has ideas

Reply:
We discussed that and decided to stick with "heuristics" for now.